verifier/src/main/scala/org/apache/celeborn/verifier/action/Operation.scala

```diff
@@ -0,0 +1,122 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.celeborn.verifier.action
+
+import java.util.concurrent.atomic.AtomicBoolean
+
+import scala.util.Random
+
+import org.apache.celeborn.common.internal.Logging
+import org.apache.celeborn.common.util.Utils
+import org.apache.celeborn.verifier.runner.RunnerContext
+import org.apache.celeborn.verifier.scheduler.SchedulerContext
+
+abstract class Operation(
+    val actionTarget: ActionTarget,
+    val updateContextBlock: RunnerContext => Unit = _ => {},
+    val interval: Long)
+  extends Serializable with Logging {
+
+  def executeOnRunner(context: RunnerContext): OperationResult
+
+  def updateSchedulerContext(context: SchedulerContext): Unit = {
+    val runContext =
+      RunnerContext(context.conf, context.runnerInfos.get(actionTarget.identity).resource)
+    updateContextBlock(runContext)
+  }
+
+  def executeCommand(command: String, block: => Unit): OperationResult = {
+    try {
+      val commandResult = Utils.runCommand(command)
```

Comment:
`Utils.runCommand` is currently not handling stderr ... which can potentially be an issue (in case some of these commands generate a bunch of stderr output) - something to address.